packages/expo-cli/src/schemes.ts

```diff
@@ -81,6 +91,55 @@ export async function getDevClientSchemeAsync(projectRoot: string): Promise<stri
   return matching;
 }
 
+async function getManagedDevClientSchemeAsync(projectRoot: string): Promise<string> {
+  const { exp, staticConfigPath, dynamicConfigPath } = getConfig(projectRoot, {
+    skipSDKVersionRequirement: true,
+  });
+  if (exp.scheme) {
+    return exp.scheme;
+  }
+
+  Log.warn(
+    '\nDev Client: No URI schemes could be found in configuration, this is required for opening the project\n'
+  );
+  if (dynamicConfigPath && !staticConfigPath) {
+    // Dynamic config (app.config.js/app.config.ts) can't be automatically updated.
+    Log.log(
+      `Add a common scheme in ${dynamicConfigPath} or provide a scheme with the ${Log.chalk.cyan(
+        '--scheme'
+      )} flag\n`
+    );
+    throw new AbortCommandError();
+  }
+
+  const { scheme } = await prompt({
+    type: 'text',
+    name: 'scheme',
+    validate: value =>
+      /^[a-z0-9-]+$/.test(value) || 'Only lowercase characters/numbers (a-z, 0-9) or hyphen (-)',
+    message: `Please choose a URI scheme for your app`,
+    initial: slugify(exp.slug, {
+      // Remove non-allowed characters and additionally hyphens, which are allowed but not idiomatic.
+      remove: /[^a-z0-9]/,
+      // Convert to lower case.
+      lower: true,
+    }),
+  });
+  if (staticConfigPath) {
```

Comment:
Maybe a bit more like this https://github.com/expo/expo-cli/blob/8b73a072907df0c1ce9dc23688f1f8980c72cd46/packages/expo-cli/src/commands/utils/modifyConfigAsync.ts#L30-L51